### PR TITLE
#13849. Fix for detecing statecurrent correctly with spoonfeeding on

### PR DESCRIPTION
**Pull request by mattw-mega**

### CODE REVIEW

File: src/megaclient.cpp [hunk 1]
```diff
@@ -4094,7 +4101,11 @@ bool MegaClient::procsc()
                             useralerts.begincatchup = true;
                         }
                     }
-                    app->catchup_result();
+
+                    if (statecurrent)
```
Comment (sergiohs84) [inline src/megaclient.cpp]:
```suggestion
                    if (statecurrent && !insca_notlast)
```
This event should be triggered only if we are actually up to date, so I think it should also check that there's no more pending actionpackets.

Comment (mattw-mega) [reply src/megaclient.cpp]: I agree - thanks very much!

File: src/megaclient.cpp [hunk 2]
```diff
@@ -4094,7 +4101,11 @@ bool MegaClient::procsc()
                             useralerts.begincatchup = true;
                         }
                     }
-                    app->catchup_result();
+
+                    if (statecurrent && !insca_notlast)
+                    {
+                        app->catchup_result();
+                    }
```
Comment (polmr) [inline src/megaclient.cpp]: If I'm not mistaken this is the same as `if (!insca_notlast)` : if `statecurrent` had evalved `false` and it was the last (i.e. !nsca_notlast == true),  the above if would have set statecurrent as `true`.   Thus, app needs to catchup if the sc is the last one here. We might want to change it for better reading.

Comment (sergiohs84) [reply src/megaclient.cpp]: That is true. We can skip to check the `statecurrent`. Please, feel free to remove it and I'll re-approve it.

**Pull request merged**